watch.go

```diff
@@ -70,11 +69,43 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			resp.Body.Close()
 		}()
 	scanning:
-		for scanner.Scan() {
+		for scanResult == nil {
 			// split event string
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
-			parts := strings.Split(scanner.Text(), "\n")
+
+			var evt []byte
+			var dat []byte
+			isPrefix := true
+			var result []byte
+
+			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
+			// we need bufio#ReadLine()
+			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
+			// so we have to add it to our result buffer.
+			evt, isPrefix, scanResult = scanner.ReadLine()
+			result = append(result, evt...)
+			result = append(result, '\n')
+
+			// 2. step: scan for the 'data:' part. Firebase returns just one 'data:'
+			// part, but the value can be very large. If we exceed a certain length
+			// isPrefix will be true until all data is read.
+			for {
```

Comment:
In L#86 we should never have isPrefix == true, because the event-type part should never be that large. Only the data part can be very large and that is why we have to call ReadLine in a loop until isPrefix == false. I did not like the idea to use isPrefix as a condition in the for-loop, as I would have to set it explicitly to isPrefix = true before the for-loop, because it is guaranteed to be false after the first and only call to ReadLine to read the event-type part. So we have to run the loop at least one time anyway. And it will stop at the right moment (when isPrefix == false).